shared_model/interfaces/iroha_internal/transaction_sequence_factory.cpp

```diff
@@ -28,28 +36,51 @@ namespace shared_model {
 
       types::BatchesCollectionType batches;
       auto insert_batch =
-          [&batches](const iroha::expected::Value<TransactionBatch> &value) {
-            batches.push_back(
-                std::make_shared<TransactionBatch>(std::move(value.value)));
-          };
+          [&batches](iroha::expected::Value<std::unique_ptr<TransactionBatch>>
+                         &value) { batches.push_back(std::move(value.value)); };
 
       validation::Answer result;
-      if (transactions.size() == 0) {
+      if (transactions.empty()) {
         result.addReason(std::make_pair(
             "Transaction collection error",
             std::vector<std::string>{"sequence can not be empty"}));
       }
       for (const auto &tx : transactions) {
+        // perform stateless validation checks
+        validation::ReasonsGroupType reason;
+        reason.first = "Transaction: ";
+        // check signatures and their validness
+        if (not boost::empty(tx->signatures())) {
```

Comment:
Revert conditional please, so that there's no `else` and code become more linear